baseplate/sidecars/live_data_watcher.py

```diff
@@ -128,25 +90,76 @@ def on_change(self, data: Optional[bytes], _znode_stat: ZnodeStat) -> None:
             if self.owner and self.group:
                 os.fchown(tmpfile.fileno(), self.owner, self.group)
             os.fchmod(tmpfile.fileno(), self.mode)
-            try:
-                json_data = json.loads(data.decode("UTF-8"))
-            except json.decoder.JSONDecodeError:
-                # If JSON fails to decode, still write the bytes data since
-                # we don't necessarily know if the the contents of the znode
-                # had to be valid JSON anyways.
-                tmpfile.write(data)
-            else:
-                # If no exceptions, we have valid JSON, and can parse accordingly.
-                data = NodeWatcher.get_data_to_write(json_data, data)
-                if data is None:
-                    logger.warning(
-                        "No data written to destination file. Something is likely misconfigured."
-                    )
-                    return
-                tmpfile.write(data)
+
+            tmpfile.write(data)
         os.rename(self.dest + ".tmp", self.dest)
 
 
+def _parse_loader_type(data: bytes) -> LoaderType:
+    try:
+        json_data = json.loads(data.decode("UTF-8"))
+    except (UnicodeDecodeError, json.decoder.JSONDecodeError):
+        logger.debug("Data is not parseable as JSON, loading as PASSTHROUGH")
```

Comment:
These are two different failure modes, feels like they should be split. When would the data not be utf-8 encoded?

Reply:
Only when it's binary data or something else shoved in to zookeeper. I found this one while running this method against all of our znodes - it only triggered for some znodes managed by other sources (flink, I _think_, but I don't remember) so we're unlikely to run in to it in our day-to-day, but I wanted to include it anyway because we've said that before